src/nvim/event/socket.c

```diff
@@ -40,17 +40,18 @@ int socket_watcher_init(Loop *loop, SocketWatcher *watcher,
       return UV_EINVAL;
     }
 
-    *host_end = '\0';
     uv_getaddrinfo_t request;
+    *host_end = '\0';
+    char *host = !strcmp(addr, "localhost") ? "127.0.0.1" : addr;
```

Comment:
Convenience, I hoped. But I really don't care. Should I leave away this commit?

I'd do anything to get this PR merged and end the back-and-forth. ;]